lib/resque/tasks.rb

```diff
@@ -41,6 +41,7 @@
   task :preload => :setup do
     if defined?(Rails) && Rails.respond_to?(:application)
       # Rails 3
+      ActiveSupport.run_load_hooks(:before_eager_load, Rails.application)
       Rails.application.eager_load!
```

Comment:
For Rails 4 it should be `Rails.application.config.eager_load_namespaces.each(&:eager_load!)`